contracts/Market.sol

```diff
@@ -963,6 +959,8 @@ contract Market is
       uint256[] memory amounts,
       address[] memory suppliers
     ) = _allocateRemovals({purchaser: purchaser, certificateAmount: amount});
+    uint256 currentPrice = _priceMultiple;
```

Comment:
I think we should move this to the top of the function to reduce future risk of the prior functions introducing changes that rely on this value